src/Data/TypeRepMap/Internal.hs

```diff
@@ -213,7 +211,13 @@ False
 True
 -}
 delete :: forall a (f :: KindOf a -> Type) . Typeable a => TypeRepMap f -> TypeRepMap f
-delete = fromTriples . deleteByFst (typeFp @a) . toTriples
+delete m
+  -- Lookups are fast, so check if we even have the element first.
+  | not (member @a m) = m
```

Comment:
Wow, that's an interesting idea! Very nice optimization :+1: 
I wonder, if we can do something like that for `insert`? For example, if the element exists, we can just update the array cell and copy the whole array instead of moving from one ordering to another.

But I guess, this will require some work, and probably be done under a separate issue :slightly_smiling_face:

Reply:
Created a separate issue for future `insert` improvements: https://github.com/kowainik/typerep-map/issues/99